### PR TITLE
extended the modem_console example. 

### PR DESCRIPTION
**Pull request by diplfranzhoepfinger**
extended the modem_console example. 

it can now use specific Features e.g. from SIM7070

### CODE REVIEW

**Review by franz-ms-muc — APPROVED** [review]

File: components/esp_modem/examples/modem_console/main/Kconfig.projbuild [hunk 1]
```diff
@@ -14,4 +14,163 @@ menu "Example Configuration"
...
+                SIMCom SIM800L is a GSM/GPRS module.
+                It supports Quad-band 850/900/1800/1900MHz.
+        config EXAMPLE_MODEM_DEVICE_BG96
+            bool "BG96"
+            help
+                Quectel BG96 is a series of LTE Cat M1/Cat NB1/EGPRS module.
+        config EXAMPLE_MODEM_DEVICE_SIM7000
+            bool "SIM7000"
+            help
+                SIM7000 is Multi-Band xxxxxxxxxxxxxxxxx
```
Comment (david-cermak) [inline components/esp_modem/examples/modem_console/main/Kconfig.projbuild:40]: Please complete the sentence

File: components/esp_modem/examples/modem_console/main/Kconfig.projbuild [hunk 2]
```diff
@@ -14,4 +14,163 @@ menu "Example Configuration"
...
+        config EXAMPLE_MODEM_DEVICE_SIM7070
+            bool "SIM7070"
+            help
+                SIM7070 is Multi-Band xxxxxxxxxxxxxxxxx
```
Comment (david-cermak) [inline components/esp_modem/examples/modem_console/main/Kconfig.projbuild:44]: same here. it doesn't have to be an exhaustive description of that device, just a complete sentence.

File: components/esp_modem/examples/modem_console/main/modem_console_main.cpp [hunk 3]
```diff
@@ -82,6 +117,17 @@ extern "C" void app_main(void)
     
     assert(dce != nullptr);
 
+    if(dte_config.uart_config.flow_control == ESP_MODEM_FLOW_CONTROL_HW)
+    {
```
Comment (david-cermak) [inline components/esp_modem/examples/modem_console/main/modem_console_main.cpp:121]: Style
```suggestion
    if(dte_config.uart_config.flow_control == ESP_MODEM_FLOW_CONTROL_HW) {
```

File: components/esp_modem/examples/modem_console/main/modem_console_main.cpp [hunk 4]
```diff
@@ -82,6 +117,17 @@ extern "C" void app_main(void)
...
+        if (command_result::OK != dce->set_flow_control(2, 2)) {
+            ESP_LOGE(TAG, "Failed to set the set_flow_control mode");
+            return;
+        }
+        ESP_LOGI(TAG, "set_flow_control OK");
+    } else {
+        ESP_LOGI(TAG, "not set_flow_control, because 2-wire mode active.");
```
Comment (david-cermak) [inline components/esp_modem/examples/modem_console/main/modem_console_main.cpp:128]: Nitpick: I would remove the `else {}` statement here, as majority of users won't setup the flow-control so this additional message might be misleading.

**Review by david-cermak — CHANGES_REQUESTED** [review]
Thanks for updating the example. This LGTM, left some nitpicks. tho.

File: components/esp_modem/examples/modem_console/main/Kconfig.projbuild (hunk 1 above)
Comment (diplfranzhoepfinger) [reply components/esp_modem/examples/modem_console/main/Kconfig.projbuild]: SIM7000 is a Multi-Band LTE-FDD and GSM/GPRS/EDGE module.

File: components/esp_modem/examples/modem_console/main/Kconfig.projbuild (hunk 2 above)
Comment (diplfranzhoepfinger) [reply components/esp_modem/examples/modem_console/main/Kconfig.projbuild]: SIM7070 is Multi-Band CAT M and NB IoT module.

**Review by david-cermak — APPROVED** [review]
Thanks for the updates!

**Comment by franz-ms-muc** [comment]
i will do same for the other Samples ?

**Comment by david-cermak** [comment]
> i will do same for the other Samples ?

Yes, please. It would be nice if you make the flow control configurable as I'm proposing in https://github.com/espressif/esp-protocols/pull/145

**Comment by tore-espressif** [comment]
@diplfranzhoepfinger  why did you add all the menuconfig option, when the application doesn't used them? e.g. `EXAMPLE_SEND_MSG`, `EXAMPLE_NEED_SIM_PIN`...

**Pull request merged**